File: helper/darwin/src/KeeWebHelper/InputCommands/CopyPasteCommand.m
#import <Cocoa/Cocoa.h>
#import <Carbon/Carbon.h>

#import "CopyPasteCommand.h"
#import "../KeyRunner.h"

@implementation CopyPasteCommand

- (id)initWithText:(NSString *)aText {
    self = [super init];
    if (self) {
        text = aText;
    }
    return self;
}

- (void)execute {
    usleep(500000);
    NSPasteboard *pasteboard = [NSPasteboard generalPasteboard];
    [pasteboard clearContents];
    [pasteboard setString:text forType:NSStringPboardType];
    usleep(500000);
    [KeyRunner keyModUpDown:kCGEventFlagMaskCommand down:true];
    [KeyRunner keyPress:0 code:kVK_ANSI_V flags:kCGEventFlagMaskCommand];
    [KeyRunner keyModUpDown:kCGEventFlagMaskCommand down:false];
    usleep(500000);
}

@end

File: helper/darwin/KeeWebHelper/KeeWebHelper/InputCommands/CopyPasteCommand.swift
import Foundation
import Cocoa
import Carbon

class CopyPasteCommand : InputCommandBase {
    let text: String
    
    init(text: String) {
        self.text = text
    }
    
    func execute() {
        usleep(500_000)
        let pasteboard = NSPasteboard.generalPasteboard();
        pasteboard.clearContents()
        pasteboard.setString(text, forType: NSStringPboardType)
        usleep(500_000)
        KeyRunner.keyUpDown(nil, code: CGKeyCode(kVK_Command), flags: nil, down: true)
        KeyRunner.keyPress(nil, code: CGKeyCode(kVK_ANSI_V), flags: CGEventFlags.MaskCommand)
        KeyRunner.keyUpDown(nil, code: CGKeyCode(kVK_Command), flags: nil, down: false)
        usleep(500_000)
    }
}
